cmd/analyzer/analyzer.go

```diff
@@ -345,22 +417,35 @@ func (a *Service) Start() {
 	ctx, cancelAnalyzers := context.WithCancel(context.Background())
 	defer cancelAnalyzers() // Start() only returns when analyzers are done, so this should be a no-op, but it makes the compiler happier.
 
-	// Start all analyzers.
+	// Start fast-sync analyzers.
+	var fastSyncWg sync.WaitGroup
+	for _, an := range a.fastSyncAnalyzers {
+		fastSyncWg.Add(1)
+		go func(an analyzer.Analyzer) {
+			defer fastSyncWg.Done()
+			an.Start(ctx)
+		}(an)
+	}
+	fastSyncAnalyzersDone := closingChannel(&fastSyncWg)
+
+	// Prepare slow-sync analyzers (to be started after fast-sync analyzers are done).
 	var wg sync.WaitGroup
-	for _, an := range a.Analyzers {
+	for _, an := range a.analyzers {
 		wg.Add(1)
 		go func(an analyzer.Analyzer) {
 			defer wg.Done()
-			an.Start(ctx)
+			// Wait for fast-sync analyzers to finish.
+			<-fastSyncAnalyzersDone
```

Comment:
This wait for the `fastSyncAnalyzersDone` closing  should be moved into the switch below (in place of the `default` case).

That way, the gorotine will also exit if the context is canceled, regardless of the state of the `fastSyncAnalyzersDone` channel.

Reply:
Ah, nice. Fast-sync analyzers all abort upon ctx.Done(), so in practice I don't think the change matters, but it is cleaner and more principled this way. Thank you!